.github/workflows/clients-dotnet.yml

```diff
@@ -1,242 +0,0 @@
-      - uses: actions/setup-dotnet@v3
-        with:
-          # Min supported and latest versions.
-          dotnet-version:
-            3.1.x
-            7.0.x
-
-      - name: Install zig
-        run: ./scripts/install_zig.sh
-
-      - name: Dotnet version
-        run: dotnet --version
-
-      - name: Build dotnet
-        working-directory: src/clients/dotnet
-        run: dotnet build --verbosity normal --configuration Release
-
-      - name: Tests
-        working-directory: src/clients/dotnet
-        run: |
-          dotnet test \
-          /p:CollectCoverage=true \
-          /p:Threshold=\"95,85,95\" \
-          /p:ThresholdType=\"line,branch,method\"
-      - name: Pack
-        working-directory: src/clients/dotnet
-        run: |
-          version=""
-          if [[ "${{ inputs.version }}" != "" ]]; then
-            version="/p:AssemblyVersion=${{ inputs.version }} /p:Version=${{ inputs.version }}"
-          fi
-          dotnet pack TigerBeetle \
-          --configuration Release \
-          $version
-
-      - name: Save nuget package
-        uses: actions/upload-artifact@v3
-        with:
-          name: nuget-artifact-${{ inputs.version }}
-          path: src/clients/dotnet/TigerBeetle/bin/Release/*.nupkg
-
-  prepare_validation_tests:
-    if: inputs.version == ''
-    needs: build
-    runs-on: ubuntu-latest
-    name: Prepare validation tests
-    steps:
-      - name: Create a simple test project
-        run: |
-          cat <<"EOF" > Program.cs
-          using System;
-          public class Program {
-            public static void Main() {
-              new TigerBeetle.Client(0, new [] {"3001"}).Dispose();
-              Console.WriteLine("SUCCESS");
-            }
-          }
-          EOF
-      - uses: actions/upload-artifact@v3
-        with:
-          name: test-project-artifact
-          path: Program.cs
-
-  run_validation_tests:
-    if: inputs.version == ''
-    needs: prepare_validation_tests
-    name: Validate OS
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ macos-latest, windows-latest]
-        # Min supported, LTS and latest versions.
-        dotnet: [ 7.0.x, 6.0.x, 3.1.x]
-    runs-on: ${{ matrix.os }}
-    steps:
-      - uses: actions/setup-dotnet@v3
-        with:
-          dotnet-version: ${{ matrix.dotnet }}
-      - uses: actions/download-artifact@v3
-        with:
-          name: nuget-artifact-${{ inputs.version }}
-          path: ./nuget
-      - name: Create test project
-        run: |
-          mkdir test-project && cd test-project
-          dotnet nuget add source ../nuget
-          dotnet new console
-          dotnet add package tigerbeetle -s ../nuget
-      - uses: actions/download-artifact@v3
-        with:
-          name: test-project-artifact
-          path: ./test-project
-      - name: Test nuget package
-        working-directory: ./test-project
-        run: dotnet run
-
-  run_validation_tests_on_containers:
```

Comment:
I am killing container tests for now and will restore then in the follow up. 

My original plan was to treat containers as OSes, and just add more lines to GHA matrix. That doesn't work, for two reasons:

* practically, on, eg, amazoncorretto, checkout action fails because the container includes neither git nor tar
* philosophically, what `zig build client_ci` tests is that the code both builds _and_ runs on host. I don't think we should constraint our build process to work on all containers. So, we fundamentally need to compile stuff in one environment, and then deploy it into the container. I imagining adding a `--container` flag to `client_ci`, which would then build on host, and then spawn docker internally to run things.

Reply:
> practically, on, eg, amazoncorretto, checkout action fails because the container includes neither git nor tar

I think we could install dependencies like that that we'd expect a user to install or have installed. I think the original idea behind testing inside them was along the lines of "does Amazon Linux ship some weird janky version of Node that breaks building TigerBeetle".

There are two separate things here:
* Using the container as a proxy for a full OS (the paragraph mentioned above) to test the build process.
* Using the container as an end user would actually use it, to test that TigerBeetle works inside of it.

Debatable if the first point is worth it!

Reply:
We have two main reasons for testing multiple OS:

- Test if our build works on Linux, MacOS, and Windows.
- Test if our binary can run on multiple OS/distro/kernels/client runtimes.

Those container tests are supposed to test the artifact already produced, I think they don't need to include any of the build dependencies, only the target client dependencies e.g. java, node, dotnet.